runtime/flang/CMakeLists.txt

```diff
@@ -498,21 +512,77 @@ else()
   set_property(TARGET flang_static PROPERTY OUTPUT_NAME flang)
 endif()
 
+set_target_properties(flang_static PROPERTIES Fortran_MODULE_DIRECTORY ${FLANG_MODULE_DIRECTORY_STATIC})
+
+if (NOT "${CMAKE_GENERATOR}" STREQUAL "Ninja")
+# State the module that the source is producing
+set_source_files_properties(
+  iso_c_bind_static.F95
+  PROPERTIES
+  OBJECT_OUTPUTS ${FLANG_MODULE_DIRECTORY_STATIC}/iso_c_binding.mod
+  )
+
+# State a dependency on the module
+set_source_files_properties(
+  ieee_arithmetic_static.F95
+  ieee_exceptions_static.F95
+  PROPERTIES
+  OBJECT_DEPENDS ${FLANG_MODULE_DIRECTORY_STATIC}/iso_c_binding.mod
+  )
+else()
+add_library(iso_c_bind OBJECT
+  iso_c_bind.F95
+)
+add_library(ieee_arithmetic OBJECT
+  ieee_arithmetic.F95
+  ieee_exceptions.F95
+)
+add_dependencies(ieee_arithmetic
+  iso_c_bind
+)
+add_dependencies(flang_static
+  ieee_arithmetic
+)
```

Comment:
These should depend of `flang1` and `flang2`

Reply:
Thanks